File: Day-1/day1.py
file = open("Day-1/input.txt", "r")

elveCalories = file.read().strip().split("\n\n")

calories = []

for calorie in elveCalories:
    items = calorie.split("\n")
    items = [eval(item) for item in items]
    calories.append(sum(items))

file.close()

calories = sorted(calories, reverse = True)
top3Calories = calories[0] + calories[1] + calories[2]

print("Solution to part 1: ", max(calories)) 
print("Solution to part 2: ", top3Calories)

File: Day-1/day1.hs
import Data.List
import Data.List.Split

main = do
    calories <- reverse . sort . readCalories <$> readFile "input.txt"
    print $ "Solution to part 1: " ++ show (head calories)
    print $ "Solution to part 2: " ++ show (sum $ take 3 calories)

readCalories :: String -> [Int]
readCalories = map (sum . map read . lines) . splitOn "\n\n"